### ISSUE TITLE
Support emojis in selectors

### ISSUE THREAD

**Issue by tomherni** [stylelint/stylelint#7281]
### What is the problem you're trying to solve?

Listen up everyone, we _totally_ need this—emojis in selectors!

When I use an emoji as a class name and assign some styles, everything works great in the browser. (It's UTF-8, after all.)

```html
<div class="🚀">Rocket goes woosh!</div>
```
```css
.🚀 {
  color: red;
}
```

But Stylelint doesn't understand what is happening, and wants my emoji to be kebab-case (which is impossible).

```
Expected class selector ".🚀" to be kebab-case
```

### What solution would you like to see?

It would be nice if there were a proper message. For example:

```
Expected class selector ".🚀" to not contain emojis
```

And, of course, more importantly—a rule to allow emojis! That way, I can deploy my beautiful emoji-riddled application to production.

Why? Because why not?

**Comment by tomherni** [comment]
When it comes to detecting emojis, simply using the following regex appears to do the trick:

```js
/\p{Extended_Pictographic}/gu.test('🚀'); // true (tested on Node 16)
```

The regex utilizes the `\p` [unicode character class escape](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Unicode_character_class_escape) with the `u` (Unicode) flag. This allows you to check if an input falls within a Unicode category.

There are several other Unicode categories, such as `Emoji` and `Emoji_Presentation`. However, the former also returns true for numbers, and the latter does not return true for all emojis.

There is also the [emoji-regex](https://www.npmjs.com/package/emoji-regex) npm package. But it may not be desirable to rely on the maintenance of a lengthy regex with emoji symbols.

**Comment by ybiquitous** [comment]
@tomherni Hi, sorry for the late response. This violation message "... be kebab-case" is defined in `stylelint-config-standard`. See below:

https://github.com/stylelint/stylelint-config-standard/blob/9d8d64daf469c6e7466594ceb003065556bcecba/index.js#L100-L105

So, you can customize the message in your config like this [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0heDcAF7ACYGD0AqDJIjPHDAXxABoQAzASwBsYA5AQwFs5EYAPTgA7MkYAM6jq4CADsGAcwQgO0AK7C+AiACcALqIwBeLAB1pGDMZB8dMaVFGX4GANqnz5y6J0BPZo3rSOgC0kLL0ckFebHZsWlCWbhgAulSJllpqMA4IJmbunjDMYDraIYxs4kECbDo2WtKOLonuuMbGAArAAKK8NnYwUAD67fTFEHJabAIAFqPkOCqpeS3AzS0WSlmibHIwjQAUooUwY1oAlIYAfBgABj0CJzZQGGDl4hhHRSVaGwAkwJ9HtpyJYMNIIDoXjIdGwAhgYMoAFb0UQ3NbmEHLZKJTGUGiheQAMW0HBqikREhosAEEkQqzynh8fgCOkalnKNi8liWHhAXl8hRZIRk8kiMJicTZIA5WVZIFMlHIQA):

Config:

```js
module.exports = {
  "extends": [
    "stylelint-config-standard"
  ],
  "rules": {
    "selector-class-pattern": [
      /\P{Extended_Pictographic}/u,
      {
        "message": (selector) => `Expected class selector "${selector}" not contain emojis`
      }
    ]
  }
}
```

Output:

```
Expected class selector ".🚀" not contain emojis
```

See also the [documentation](https://stylelint.io/user-guide/configure#message).
